src/main/java/com/pet/bankservice/service/impl/ExchangeHostRateFetcher.java

```diff
@@ -0,0 +1,52 @@
+package com.pet.bankservice.service.impl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.pet.bankservice.entity.Currency;
+import com.pet.bankservice.entity.dto.CurrencyExchangeDto;
+import com.pet.bankservice.exception.ApiUnavailableException;
+import com.pet.bankservice.service.ExchangeRateFetcher;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import lombok.AllArgsConstructor;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class ExchangeHostRateFetcher implements ExchangeRateFetcher {
+    private static final String API_PROP_NAME = "exchangerate.api.url";
+    private static final String DATE_FORMAT_PROP_NAME = "exchangerate.date.format";
+    private final Environment env;
+    private final ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
in Spring this may be injected through the constructor without even creating a new object

Reply:
done